### ISSUE TITLE
Autocomplete sometimes fails in fish shell

### ISSUE THREAD

**Issue by zmitchell** [flox/flox#2161]
On `flox` version 1.3.2-gb91c3f1 I was typing `flox activate -v` when I got this error:
```console
❯ flox activate -vcomplete: too many arguments

/var/folders/qn/77rf0syj2s7djp588bzp5vkm0000gn/T//.psub.pzYGY4Pmvo (line 1):
complete -c flox -f --long-option  -d "Positional only items after this token"
^
from sourcing file /var/folders/qn/77rf0syj2s7djp588bzp5vkm0000gn/T//.psub.pzYGY4Pmvo
        called on line 6 of file ~/.nix-profile/share/fish/vendor_completions.d/flox.fish
from sourcing file ~/.nix-profile/share/fish/vendor_completions.d/flox.fish

(Type 'help complete' for related documentation)
v                  v
```

I've seen completions errors of one kind or another from time to time, and haven't noticed a pattern yet, but there seems to be an issue with completions in `fish`.

**Comment by zmitchell** [comment]
Another incident after updating to 1.3.3, this time typing out `flox edit`:
```
❯ FLOX_FEATURES_BUILD=true flox complete: too many arguments

~/.nix-profile/share/fish/vendor_completions.d/flox.fish (line 10):
complete --no-files --command flox--arguments '(_bpaf_dynamic_completion)'
^
from sourcing file ~/.nix-profile/share/fish/vendor_completions.d/flox.fish

(Type 'help complete' for related documentation)
e                                 edit
```

**Comment by pacak** [comment]
I made some fixes to fish shell completions recently around here, is it before or after?

 https://github.com/pacak/bpaf/pull/389/commits/baec2fd64ed97dee150204ab40827a44ee1a6d20

If after - you need to regenerate completion files.

**Comment by mkenigs** [comment]
I think this is after that commit based on the contents of my `/usr/local/share/fish/vendor_completions.d/flox.fish`
```
flox [mkenigs/default] ~  > flox --version
1.3.3

flox [mkenigs/default] ~  > fish --version
fish, version 3.7.1

flox [mkenigs/default] ~  > fish
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
matthew@Matthews-MacBook-Pro-2 ~> flox ecomplete: too many arguments

/usr/local/share/fish/vendor_completions.d/flox.fish (line 10):
complete --no-files --command flox--arguments '(_bpaf_dynamic_completion)'
^
from sourcing file /usr/local/share/fish/vendor_completions.d/flox.fish

(Type 'help complete' for related documentation)

matthew@Matthews-MacBook-Pro-2 ~> cat /usr/local/share/fish/vendor_completions.d/flox.fish
function _bpaf_dynamic_completion
    set -l current (commandline --tokenize --current-process)
    set -l tmpline --bpaf-complete-rev=9 $current[2..]
    if test (commandline --current-process) != (string trim (commandline --current-process))
        set tmpline $tmpline ""
    end
    eval $current[1] \"$tmpline\"
end

complete --no-files --command flox--arguments '(_bpaf_dynamic_completion)'
```

**Comment by mkenigs** [comment]
I think there's just a missing space between flox and `--arguments`

**Comment by mkenigs** [comment]
We should probably add a test for our completions

**Comment by pacak** [comment]
> I think there's just a missing space between flox and --arguments

Indeed. I'll make a fix for that and release a new `bpaf` version. And will try to figure out how it worked for me locally...

**Comment by mkenigs** [comment]
Amazing thank you!

**Comment by pacak** [comment]
0.9.15 should be out, https://github.com/pacak/bpaf/pull/392

**Comment by mkenigs** [comment]
That was fast :) Thanks!

**Comment by ysndr** [comment]
fwiw we are building with 0.9.15 since https://github.com/flox/flox/pull/2244

**Comment by zmitchell** [comment]
Just FYI, even though I can build `main` locally, my `nix-darwin` upgrade failed:
```
       > src/realisepkgs/realise.cc:60:9: error: no member named 'force' in 'nix::eval_cache::CachedEvalError'
       >       e.force();
       >       ~ ^
```

**Comment by zmitchell** [comment]
`nix shell github:flox/flox` works though...

**Comment by mkenigs** [comment]
> Just FYI, even though I can build `main` locally, my `nix-darwin` upgrade failed:
> 
> ```
>        > src/realisepkgs/realise.cc:60:9: error: no member named 'force' in 'nix::eval_cache::CachedEvalError'
>        >       e.force();
>        >       ~ ^
> ```
> 
> I was attempting to upgrade to `main` to see if the issue was resolved.

Looks like mismatched developer environment and source maybe? I would expect to see that error if I built source from after https://github.com/flox/flox/pull/2231 against a Nix from before https://github.com/flox/flox/pull/2231

**Comment by zmitchell** [comment]
I get the same error on my NixOS machine as well. I'm not sure what you mean by "developer environment" because these were both just setting the flake ref to `github:flox/flox` in the flake that configures my laptop and desktop. Both were applied with a `nixos-rebuild switch` or `darwin-rebuild switch`. If we have users who have installed via Nix they may hit the same error.

**Comment by dcarley** [comment]
I can't reproduce the original problem to see whether it's fixed. Am I missing something?
```
% nix shell 'nixpkgs#fish' 'github:flox/flox/v1.3.2#flox' --command fish
warning: input 'crane' has an override for a non-existent input 'nixpkgs'
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
dcarley@mba15 ~/dotfiles (master)> fish --version
fish, version 3.7.1
dcarley@mba15 ~/dotfiles (master)> flox --version
1.3.2-gb91c3f1
dcarley@mba15 ~/dotfiles (master)> flox activate -v -<TAB>
-h  --help                        (Prints help information)  -v  --verbose  (Increase logging verbosity. Invoke multiple times for increasing detail)
-q  --quiet  (Silence logs except for errors. flox Options)  --version                                                           (Print flox version)
```

> Just FYI, even though I can build `main` locally, my `nix-darwin` upgrade failed:

This was solved with a `nix flake lock --update-input flox` ([internal Slack](https://flox-dev.slack.com/archives/C055P1JCP5L/p1730318707167339)).

**Comment by mkenigs** [comment]
Confirmed we have the fix with 1.3.4 in `/usr/local/share/fish/vendor_completions.d/flox.fish`
